scripts/gen_vimdoc.py

```diff
@@ -96,19 +102,73 @@
 
 text_width = 78
 indentation = 4
+SECTION_SEP = '=' * text_width
+
 script_path = os.path.abspath(__file__)
 base_dir = os.path.dirname(os.path.dirname(script_path))
 out_dir = os.path.join(base_dir, 'tmp-{target}-doc')
 filter_cmd = '%s %s' % (sys.executable, script_path)
 msgs = []  # Messages to show on exit.
 lua2dox = os.path.join(base_dir, 'scripts', 'lua2dox.lua')
 
-CONFIG = {
-    'api': {
-        'mode': 'c',
-        'filename': 'api.txt',
+
+SectionName = str
+
+Docstring = str  # Represents (formatted) vimdoc string
+
+FunctionName = str
+
+
+@dataclasses.dataclass
+class Config:
+    """Config for documentation."""
+
+    mode: Literal['c', 'lua']
+
+    filename: str
+    """Generated documentation target, e.g. api.txt"""
```

Comment:
Is this standard practice to put the docstring _below_ the field? I've written a lot of Python and have never seen this before. It threw me off at first.

Reply:
Yes, in this way LSP (pyright) can recognize docstrings on the field. If it were `above`, it will go to the wrong adjacent field.

<img width="807" alt="image" src="https://github.com/neovim/neovim/assets/1009873/b6a6d000-66bd-4312-9788-8527eb7d93f0">

Reply:
Pyright is not a language server ;)

Reply:
>Yes, in this way LSP (pyright) can recognize docstrings on the field

Huh, TIL.